src/test/coll.rs

```diff
@@ -1049,3 +1053,258 @@ async fn collection_generic_bounds() {
         .collection(function_name!());
     let _result = coll.insert_one(Bar {}, None).await;
 }
+
```

Comment:
In addition to these tests we should also make sure we're implementing the [prose tests](https://github.com/mongodb/specifications/blob/master/source/enumerate-indexes.rst#tests) described at the end of the enumerate indexes spec; these tests already encompass some of what's described but we should make sure we're covering all of those cases.

Reply:
Okay - I had created a ticket for this but I think it makes sense to do this on this PR for the sake of correctness, so I'll take care of that.

Reply:
oh didn't realize there was already a ticket for it, but feel free to do here if you think that's the right move

Reply:
After looking into it closely it seems like it's really a 15 line modification to the current prose testing, since I had handled almost everything in `index_management_lists`. Gonna make that change and merge that here.